aten/src/ATen/cuda/CUDAMultiStreamGuard.h

```diff
@@ -16,32 +16,22 @@ class CUDAMultiStreamGuard final {
   /// Calls `set_stream` on each of the streams in the list.
   /// This may be useful if you need to set different streams
   /// for different devices.
-  explicit CUDAMultiStreamGuard(ArrayRef<CUDAStream> streams) : CUDAMultiStreamGuard() {
+  explicit CUDAMultiStreamGuard(ArrayRef<CUDAStream> streams) {
+    original_streams_.reserve(streams.size());
     for (const auto& s : streams) {
+      original_streams_.push_back(getCurrentCUDAStream(s.device_index()));
       setCurrentCUDAStream(s);
     }
   }
 
-  CUDAMultiStreamGuard() {
-    const size_t device_count = getNumGPUs();
-    original_streams_.reserve(device_count);
-    for (size_t device = 0; device < device_count; ++device) {
-      original_streams_.push_back(getCurrentCUDAStream(device));
-    }
-  }
-
-  CUDAMultiStreamGuard(const CUDAGuard&) = delete;
```

Comment:
curious, why existing deleted assignment and move constructors took a `CUDAGuard` instead of a `CUDAMultiStreamGuard`?

Reply:
I suspect this came from a botched copy-paste.